Client/AccessoryViewProvider.swift

```diff
@@ -0,0 +1,144 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+import Shared
+
+class AccessoryView: UIView, Themeable {
+    private struct AccessoryViewUX {
+        static let toolbarHeight: CGFloat = 50
+    }
+
+    var themeManager: ThemeManager
+    var themeObserver: NSObjectProtocol?
+    var notificationCenter: NotificationProtocol
+    var showCreditCard = false
+    let target: TabWebView
+    let throttler = Throttler(seconds: 1.0)
+
+    private var toolbar: UIToolbar = .build { toolbar in
+        toolbar.sizeToFit()
+    }
+
+    lazy private var previousButton = UIBarButtonItem(image: UIImage(systemName: "chevron.up"),
+                                                      style: .plain,
+                                                      target: target,
+                                                      action: nil)
+
+    lazy private var nextButton = UIBarButtonItem(image: UIImage(systemName: "chevron.down"),
+                                                  style: .plain,
+                                                  target: target,
+                                                  action: nil)
+
+    lazy private var doneButton = UIBarButtonItem(title: "Done",
+                                                  style: .done,
+                                                  target: target,
+                                                  action: nil)
```

Comment:
Actions aren't working here either, how do you plan on adding the next and previous actions along with done button. We should definitely add comments with stubs as this isn't full solution

Reply:
**Note**: Your button is marked lazy private without proper initialization so adding a selector here will cause a crash as you aren't initializing it. Make sure to initialize it :) 

Example: You can add something like this in AccessoryView. This isn't final piece but should give you an idea on stubs.

```
    var previousClosure: (() -> Void)?
    var nextClosure: (() -> Void)?
```

Then hook them with button after proper initialization. Just adding selector will cause a crash.
```
    lazy private var previousButton = UIBarButtonItem(image: UIImage(systemName: "chevron.up"),
                                                      style: .plain,
                                                      target: target,
                                                      action: #selector(previousAction))
```

```
    @objc
    public func nextAction() {
        nextClosure?()
    }
 ```
 
 Lastly hook them up via legacy delegate by adding the following to `TabWebView`
 
 ```
     override init(frame: CGRect, configuration: WKWebViewConfiguration) {
        super.init(frame: frame, configuration: configuration)
        accessoryView = AccessoryView(for: self)
        accessoryView?.nextClosure = {
            self.delegate?.tabWebViewFocusNextField(self)
        }

        accessoryView?.previousClosure = {
            self.delegate?.tabWebViewFocusNextField(self)
        }
    }
 ```